Makefile

```diff
@@ -53,7 +58,15 @@ LDFLAGS='-extldflags "-static"'
 controller-bin:
 	@echo '********'
 	@echo 'Compiling controller binary'
-	docker run -it --rm -v $(REPO_ROOT):$(WORK_DIR):Z -w $(WORK_DIR) -e GOOS=$(GOOS) -e GOARCH=$(ARCH) -e CGO_ENABLED=$(CGO_ENABLED) $(BUILD_IMAGE) go build -a -ldflags $(LDFLAGS) -o $(WORK_DIR)/bin/$(CONTROLLER_BIN) $(WORK_DIR)/cmd/controller/controller.go
+	$(eval DOCKER_CMD_COUNT=$(shell docker ps | wc -c))
+	export DOCKER_CMD_COUNT
+	@if [ $(DOCKER_CMD_COUNT) -eq 0 ] || [ $(USE_DOCKER) -eq 0 ]; then \
```

Comment:
it _may_ be cleaner to use make if/else blocks rather than in shell.

Reply:
I've tried, and the `ifeq else endif` blocks didn't work, they errored and/or they never evaluated the vars correctly. that is why I moved to the shell, but probably the bigger issue I didn't like is there is no complex conditional support (if AND/OR) so we would have to nest our ifs